### ISSUE TITLE
Interpreting decimal points

### ISSUE THREAD

**Issue by gordonsmithwkuedu** [numbas/Numbas#611]
Numbas is marking an answer with no leading zero in a decimal point as incorrect in a "question with units" problem (Entropy generation through a wall).

i.e., "0.823 W/degK" would be marked correct, but ".823 W/degK" would be marked wrong.

**Comment by christianp** [comment]
That's the intended behaviour. Are you teaching students to omit the leading zero?

**Comment by gordonsmithwkuedu** [comment]
It’s not a question of teaching them to omit (because I’m not ☺ ), but the confusion that occurs because (like it or not) they are technically equivalent.  Is there a way to put in a “correct number but incorrectly formatted” sort of flag to help guide them to the proper format without their reading it that they produced an incorrect end value?

**Comment by christianp** [comment]
Ah, I hadn't realised that it wasn't giving a good error message when the number part of your answer is invalid.
I've added a step at the start of the marking algorithm, so it fails and gives an error "You did not enter a valid number" if the scalar part couldn't be parsed.
I've made a separate issue, #616, about allowing numbers without a leading zero. It's complicated, and should be done in a consistent way across all the places numbers are parsed.
